packages/blockchain-wallet-v4-frontend/src/components/Lockbox/AppManager/index.js

```diff
@@ -0,0 +1,247 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+import { FormattedHTMLMessage } from 'react-intl'
+
+import {
+  BlockchainLoader,
+  Button,
+  Image,
+  Text
+} from 'blockchain-info-components'
+import { actions, selectors } from 'data'
+import * as Lockbox from 'services/LockboxService'
+import LockboxAppManager from './template'
+import { Remote } from 'blockchain-wallet-v4'
+
+const Wrapper = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+`
+const ConnectStep = styled.div`
+  text-align: center;
+  & > :last-child {
+    margin: 10px 0;
+  }
+`
+const Loader = styled(BlockchainLoader)`
+  margin: 20px;
+`
+const ContinueButton = styled(Button)`
+  margin-top: 12px;
+`
+const ConnectInstructions = styled(Text)`
+  text-align: center;
+  padding: 5px;
+  margin: 30px 0 36px;
+`
+const AllowManagerText = styled(Text)`
+  margin: 20px 0;
+  text-align: center;
+`
+const BtcRequiredText = styled(Text)`
+  margin-top: 8px;
+  text-align: center;
+`
+
+const getKeyByValue = value => {
+  return Object.keys(Lockbox.constants.supportedApps).find(
+    key => Lockbox.constants.supportedApps[key] === value
+  )
+}
+
+class LockboxAppManagerContainer extends React.PureComponent {
+  state = {}
+
+  componentDidMount () {
+    this.props.lockboxActions.initializeAppManager(this.props.deviceIndex)
+  }
+
+  componentWillUnmount () {
+    this.props.lockboxActions.resetAppChangeStatus()
+  }
+
+  onAppInstall = (appName, coin) => {
+    this.setState({
+      [appName]: {
+        status: 'Updating',
+        changeType: 'Installing',
+        appName
+      }
+    })
+    this.props.lockboxActions.installApplication(coin)
+  }
+
+  onAppUninstall = appName => {
+    this.setState({
+      [appName]: {
+        status: 'Updating',
+        changeType: 'Uninstalling',
+        appName
+      }
+    })
+    this.props.lockboxActions.uninstallApplication(appName)
+  }
+
+  render () {
+    const { appChangeStatus, appVersionInfos, connection } = this.props
+
+    appChangeStatus.cata({
+      Success: val => {
+        // install/uninstall APIs use different keys for appName
+        const AppName = Lockbox.constants.supportedApps[val.appName]
+          ? Lockbox.constants.supportedApps[val.appName]
+          : val.appName
+        this.props.lockboxActions.resetAppChangeStatus()
+        this.setState({
```

Comment:
I think we should consider changing this behaviour as firing actions, doing setState, timeout => setState is a bad idea in render. we can use componentDidUpdate or getDerivedStateFromProps

Reply:
Moved the entire `cata` into componentDidUpdate.  Let me know if this is sufficient.